apps/studio/src/components/CoreTabs.vue

```diff
@@ -593,6 +599,17 @@ export default Vue.extend({
       this.tableExportOptions = options
       this.showExportModal = true
     },
+    async alterTable(options: { changes: AlterTableSpec, table: TableOrView, onError?: (e: any) => void, onSuccess?: () => void }) {
+      try {
+        await this.connection.alterTable(options.changes)
+        await this.$store.dispatch('updateTableColumns', options.table)
+        options.onSuccess?.()
+        this.trigger(AppEvent.tableAltered, options.table)
+      } catch (e) {
+        log.error(e);
+        options.onError?.(e)
```

Comment:
It's not a good idea to pass functions around on the event bus. It's really just for data.

I don't really see the reason for `alterTable` to be pushed up to CoreTabs, why can't this stay in `TableSchema`?